trinity/extensibility/plugin_manager.py

```diff
@@ -219,14 +187,15 @@ def shutdown_blocking(self) -> None:
 
         self._logger.info("Shutting down PluginManager with scope %s", type(self._scope))
 
-        for plugin in self._started_plugins:
+        for plugin in self._plugin_store:
 
-            if not isinstance(plugin, BaseSyncStopPlugin):
+            if not isinstance(plugin, BaseSyncStopPlugin) or not plugin.running:
                 continue
 
             try:
                 self._logger.info("Stopping plugin: %s", plugin.name)
                 plugin.stop()
+                plugin.running = False
```

Comment:
This seems wrong for something external to the plugin to manage its state.  Shouldn't this be handled by the `plugin.stop` method?

Reply:
Mmh...well, since `stop` is implemented by each individual plugin that would mean that plugin authors either need to remember to call `super().stop()` (bad because if they forget it, they are basically leaving the `PluginManager` thinking that the plugin would still be running when in reality it isn't.

Alternatively we do the same as we do with the `start` / `boot` pair which means there will be another method (`shutdown` ?) which sets `running=False` and then delegates to `stop()`. At least, that would be symmetric to the `start / boot` pair. 

However setting `plugin.running` from the manager lets us avoid this extra indirection and I'm personally ok with having the `PluginManager` make the flip here. Your call.

Reply:
I'm generally a fan of the model you proposed with `stop/shutdown`.  One is internal, only accessed by the manager and handles the orchestration like setting flags, triggering `asyncio.Event`s, the other is the hook for the user to implement the logic.  We've been using this with `BaseService` for a while and I think it's working out well.

Reply:
Alright, will refactor that previous to merging this.

Reply:
@pipermerriam this https://github.com/ethereum/py-evm/pull/1331/commits/5ee9b53b73152eb0194bb9bf6660b39d73031028